src/js/sonifier.js

```diff
@@ -0,0 +1,260 @@
+    "use strict";
+
+    fluid.defaults("floe.chartAuthoring.sonifier", {
+        gradeNames: ["floe.chartAuthoring.totalRelaying", "floe.chartAuthoring.sonifier.unitDivisorSonificationStrategy"],
+        components: {
+            textToSpeech: {
+                type: "fluid.textToSpeech",
+                options: {
+                    model: {
+                        utteranceOpts: {
+                            "lang": "en-US"
+                        }
+                    }
+                }
+
+            },
+            enviro: {
+                type: "flock.enviro"
+            },
+            synth: {
+                type: "floe.chartAuthoring.electricPianoBand",
+                options: {
+                    components: {
+                        scheduler: {
+                            type: "flock.scheduler.async",
+                            options: {
+                                synthContext: "floe.chartAuthoring.electricPianoBand"
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        model: {
+            // dataSet:,
+            // sonifiedData:
+            // sonificationQueue:
+            // currentlyPlayingData: a data object, or false if no data playing
+            // Supplied by relaying in floe.chartAuthoring.totalRelaying grade
+            // total: {
+            //     value: number,
+            //     percentage: number
+            // }
+            isPlaying: false,
+            isTextToSpeechSupported: {
+                expander: {
+                    func: "fluid.textToSpeech.isSupported"
+                }
+            }
+        },
+        modelListeners: {
+            dataSet: {
+                funcName: "{that}.sonifyData"
+            },
+            sonificationQueue: {
+                funcName: "{that}.beginSonificationQueue"
+            }
+        },
+        // options that control playback behaviour
+        playbackOptions: {
+            // Additional gap of silence between individual data points when
+            // playing a sequence, in seconds
+            gapDuration: 1,
+            // Zoom factor - affects the durationConfigs of the synth when
+            // played. Higher numbers = slower playback
+            zoom: 1
+        },
+        invokers: {
+            "stopSonification": {
+                funcName: "floe.chartAuthoring.sonifier.stopSonification",
+                args: "{that}"
+            },
+            "beginSonificationQueue": {
+                funcName: "floe.chartAuthoring.sonifier.processSonificationQueue",
+                args: [0, true, "{that}"]
+            },
+            "playDataAndQueueNext" : {
+                funcName: "floe.chartAuthoring.sonifier.playDataAndQueueNext",
+                args: ["{that}"]
+            },
+            // Calls the default sonification strategy
+            "defaultSonificationStrategy": "{that}.unitDivisorStrategy(10)"
```

Comment:
Does it make sense to define the number ```10``` as a component option so it's user configurable?